src/validation.cpp

```diff
@@ -2645,6 +2645,7 @@ bool CChainState::ActivateBestChainStep(BlockValidationState& state, CBlockIndex
 {
     AssertLockHeld(cs_main);
     if (m_mempool) AssertLockHeld(m_mempool->cs);
+    if (m_stempool) AssertLockHeld(m_stempool->cs);
```

Comment:
not sure if this is needed, 'MaybeUpdateMempoolForReorg' contains an assert for stempool already

Reply:
Yeah agree with @barrystyle here, seems redundant.

Reply:
@SmartArray ,

Are you able to remove the redundant asserts?

Reply:
I see where you are coming from. The signature of `MaybeUpdateMempoolForReorg`requires callers to assert that locks are held already. Super useful for static analysis.

This is also the reason the Bitcoin Developers added a check for the mempool mutex (L2647) although it is checked again in `MaybeUpdateMempoolForReorg`.